### PR TITLE
Sample App

### PR DESCRIPTION
**Pull request by AnanyaBarthakur1**
Basic app with msal auth, msGraph, fluentUI

### CODE REVIEW

File: apps/sample-app/index.html
Comment (MarcinJuraszek) [inline apps/sample-app/index.html:16]: Same here, can you make sure `title` matches what the app is meant to be?

File: apps/sample-app/package.json
Comment (MarcinJuraszek) [inline apps/sample-app/package.json:4]: should be `Microsoft Teams`

File: apps/sample-app/src/App.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/App.tsx:28]: should probably use `React.useCallback` instead.

File: apps/sample-app/src/App.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/App.tsx:35]: This is not used, to can you just remove it instead of working around an ESLint error with a disable-next-line?

File: apps/sample-app/src/App.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/App.tsx:42]: You have two separate places where `app.initialize()` is called. There should only be one.

File: apps/sample-app/src/App.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/App.tsx:37]: the initial theme should be taken from `getContext` and later optionally updates via `registerOnThemeChangeHandler`

File: apps/sample-app/src/App.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/App.tsx:61]: If app initialization fails, you probably want to do something more than just print it to console.

File: apps/sample-app/src/AuthApp.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/AuthApp.tsx:16]: Would be great to handle errors here too

File: apps/sample-app/src/PageLayout.jsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/PageLayout.jsx:13]: you're using `SignInButton` as if it was a react component, but I don't think it is.

File: apps/sample-app/src/components/Profile.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/components/Profile.tsx:34]: Do we need to do this initialization twice, once for messages and once for user info?

File: package.json
Comment (MarcinJuraszek) [inline package.json:107]: These should be in the package.json for sample app, not the top-level one.

Comment (aeunms) [reply package.json:107]: I looked through the other apps and they did have their own dependencies and devDependencies, should I create a ticket to get that cleaned up for all apps?

Comment (aeunms) [reply package.json:107]: There were just few if that helps

Comment (MarcinJuraszek) [reply package.json:107]: hmm, I don't think it matters for `devDependencies`, and for `dependencies` it looks like it used to be an empty set `{}` and you're adding first items here. So not sure if there's work needed to clean this up for other apps. Just make sure you move the fluentui dependencies you're adding to the sample app's package.json.

File: apps/sample-app/src/components/Profile.tsx
Comment (TrevorJoelHarris) [inline apps/sample-app/src/components/Profile.tsx:39]: 2 as a suffix on response isn't necessary, this response is in a different scope than the one on line 26

File: apps/sample-app/src/components/TokenFetch.tsx
Comment (TrevorJoelHarris) [inline apps/sample-app/src/components/TokenFetch.tsx:26]: nit: Patterns like this where you check a boolean and then print UI and then on the next line check the opposite and print different UI can sometimes be a little easier to read and maintain if you write them like this:
```suggestion
      {!accessToken ? <><p>Fetching access token...</p><Spinner size={SpinnerSize.large} /></> : <p>...</p>}
```

File: apps/sample-app/src/index.tsx
Comment (TrevorJoelHarris) [inline apps/sample-app/src/index.tsx:12]: We should construct msalInstance inside of the if block at line 18 so we only initialize this when we're actually going to use it

File: package.json
Comment (TrevorJoelHarris) [inline package.json:108]: Probably we shouldn't use northstar and "normal" react. We should just pick one flavor of react and use that (and I would vote for "not northstar")

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

**Review by TrevorJoelHarris — COMMENTED** [review]
Marking as comment and not Request changes because I will be out next week and I don't want to hold up your check in.

File: apps/sample-app/README.md
Comment (MarcinJuraszek) [reply apps/sample-app/README.md:1]: Let's track that for future changes, and not block this PR on it

**Comment by AE-MS** [comment]
>   "name": "teams-js-monorepo",

This is probably not correct.

---
Refers to: package.json:2 in 8ec63e2. [](commit_id = 8ec63e25abec1da39555c56281aa12c1a040e5f3, deletion_comment = False)

File: apps/sample-app/public/manifest.json
Comment (AE-MS) [inline apps/sample-app/public/manifest.json:8]: >   "name": "Ananya Barthakur", [](http://example.com/codeflow?start=1&length=30)

This may not be the right name to put here. Consider "Microsoft"?

File: apps/sample-app/src/components/Button.tsx
Comment (AnanyaBarthakur1) [reply apps/sample-app/src/components/Button.tsx:5]: Yes, this "SignInButton" component serves as an automatic sign in for the user and is not meant to show an actual button. This is to improve the user experience by removing the need to press another button once the authentication page opens up

File: apps/sample-app/src/App.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/App.tsx:34]: I don't think React guarantees the order in which `useEffect` callbacks are executed. So I'm wondering if there's a potential for this one to run before the one on line 14 is done, which would result in API calls being made before the app is properly initialized. 

Can you move all this to a single useEffect instead?

File: apps/sample-app/public/manifest.json
Comment (MarcinJuraszek) [inline apps/sample-app/public/manifest.json:51]: Can you please track an item for the future to enable `showLoadingIndicator` for the sample app?

File: apps/sample-app/src/App.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/App.tsx:27]: hardcoding this to `https://localhost:4003` means if we were to deploy it somewhere (like Azure) it would stop working. Can you use relative paths instead? Using `URL` to get one might be a good idea: https://developer.mozilla.org/en-US/docs/Web/API/URL/URL. I haven't tried it, but something like `(new URL('/?auth=1')).toString()` might work

File: apps/sample-app/src/components/TokenFetch.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/components/TokenFetch.tsx:22]: You most likely don't want to call this again, instead you should attach the `catch()` to the initial `acquireTokenSilent` promise:

```
instance.acquireTokenSilent(request).then(response => {
      setAccessToken(response.accessToken);
      authentication.notifySuccess(response.accessToken);
    }).catch(response => { ... });
```

File: apps/sample-app/src/components/Profile.tsx
Comment (MarcinJuraszek) [inline apps/sample-app/src/components/Profile.tsx:32]: yes, please move this to a utility function

File: apps/sample-app/src/components/utils.tsx
Comment (aeunms) [inline apps/sample-app/src/components/utils.tsx:26]: What happens if currDate is the last day of the month? e.g. January 31st?

Comment (AnanyaBarthakur1) [reply apps/sample-app/src/components/utils.tsx:26]: Oh right! Thanks for pointing this out to me. I have changed the code to use tomorrow.setDate(current.getDate() + 1); (getDate will get the right 'day number', and setDate will get the month and year accordingly)

**Review by aeunms — DISMISSED** [review]
🚢  Great job getting so many great changes in in such a short time!! :)

File: apps/sample-app/src/components/Calendar.tsx
Comment (TrevorJoelHarris) [inline apps/sample-app/src/components/Calendar.tsx:11]: You may want to replace this with a placeholder value instead of using your real address.

Comment (TrevorJoelHarris) [reply apps/sample-app/src/components/Calendar.tsx:11]: Something like 'emailaddress@company.com'

File: apps/sample-app/src/components/PeopleAvatars.tsx
Comment (TrevorJoelHarris) [inline apps/sample-app/src/components/PeopleAvatars.tsx:80]: Intentional space ahead of the word Video?

Comment (TrevorJoelHarris) [reply apps/sample-app/src/components/PeopleAvatars.tsx:80]: Similar with the space ahead of {c} on line 85

File: apps/sample-app/src/components/PeopleAvatars.tsx
Comment (TrevorJoelHarris) [inline apps/sample-app/src/components/PeopleAvatars.tsx:107]: I don't think this div is needed, AvatarExample starts with a div

File: apps/sample-app/src/components/Profile.tsx
Comment (TrevorJoelHarris) [inline apps/sample-app/src/components/Profile.tsx:25]: probably good to remove this console.log unless it's giving consumers of these samples some info they need to understand what's going on

File: apps/sample-app/src/components/TokenFetch.tsx
Comment (TrevorJoelHarris) [inline apps/sample-app/src/components/TokenFetch.tsx:33]: <p> tags should be <Text> from fluent

File: apps/sample-app/src/components/utils.tsx
Comment (TrevorJoelHarris) [inline apps/sample-app/src/components/utils.tsx:24]: this is very minor, but having this return an interface like:

```
interface DateCollection {
  today: string;
  tomorrow: string;
}
```

Would be more user friendly and easier to read. I'll let you decide whether that's needed in this sample app.

**Review by TrevorJoelHarris — APPROVED** [review]

**Comment by aeunms** [comment]
Merging to get the test app in and make deploying a lot easier. The changes suggested by @TrevorJoelHarris will be integrated in a future PR!

**Review by aeunms — APPROVED** [review]

**Review by aeunms — APPROVED** [review]

**Review by AE-MS — APPROVED** [review]

**Pull request merged**
